File: HRMS/Models/EmployeePositionUpdate.cs
using HRMS.Core.Models.Enums;
using System.ComponentModel.DataAnnotations;

namespace HRMS.Models;

public class EmployeePositionUpdate
{
    [Required]
    public int EmployeeId { get; set; }

    [Required]
    public int PositionId { get; set; }

    public DateTime StartDate { get; set; }

    public DateTime? EndDate { get; set; }

    public decimal Salary { get; set; }

    public int VacationDays { get; set; }

    public EmploymentType EmploymentType { get; set; }

    public string WorkingHours { get; set; } = string.Empty;
}

File: HRMS.Desktop/lib/models/requests/employee_position_update.dart
import 'package:json_annotation/json_annotation.dart';

import '../enums/employment_type.dart';

part 'employee_position_update.g.dart';

@JsonSerializable()
class EmployeePositionUpdate {
  int employeeId;
  int positionId;
  DateTime startDate;
  DateTime? endDate;
  double salary;
  int vacationDays;
  EmploymentType employmentType;
  String workingHours;

  EmployeePositionUpdate(
    this.employeeId,
    this.positionId,
    this.startDate,
    this.endDate,
    this.salary,
    this.vacationDays,
    this.employmentType,
    this.workingHours,
  );

  factory EmployeePositionUpdate.fromJson(Map<String, dynamic> json) =>
      _$EmployeePositionUpdateFromJson(json);

  Map<String, dynamic> toJson() => _$EmployeePositionUpdateToJson(this);
}
